cmd/integration_test.go

```diff
@@ -306,85 +328,109 @@ func TestSmokescreenIntegration(t *testing.T) {
 
 	var testCases []*TestCase
 
+	// This generates all the permutations for the common test cases
```

Comment:
A little more commentary here would be helpful, especially since inside the nested loop we decide that some of the test cases aren't interesting/meaningful.

Reply:
Also the block of `*Domain` variables above merits a comment for readability.